modules/gapi/misc/python/test/test_gapi_sample_pipelines.py

```diff
@@ -522,5 +375,288 @@ def test_custom_op_goodFeaturesToTrack(self):
                                       np.array(actual, dtype=np.float32).flatten(), cv.NORM_INF))
 
 
+    def test_invalid_op(self):
```

Comment:
With fixed absolute path I'm observing sporadic crashed on this test case or tests cleanup.
Stacktrace is not really useful (python-only runtime code).

```
Ran 17 tests in 0.064s

OK
Segmentation fault (core dumped)
```

```
test_invalid_op (test_gapi_sample_pipelines.gapi_sample_pipelines) ... Segmentation fault (core dumped)
```

Launch command example:
```
OPENCV_PYTEST_FILTER=test_gapi_sample* ./setup_vars.sh python3 ../opencv/modules/python/test/test.py -v --repo ../opencv
```

Please confirm.

Reply:
Works fine for me:

```
atalaman@nnlicv179:~/workspace/opencv-python/build$ OPENCV_PYTEST_FILTER=test_gapi_sample* ./setup_vars.sh python3 ../../opencv-python/modules/python/test/test.py -v --repo ../../opencv-python
Setting vars for OpenCV 4.5.2-dev
Append PYTHONPATH: /home/atalaman/workspace/opencv-python/build/python_loader
Testing OpenCV 4.5.2-dev
Local repo path: ../../opencv-python
Tests filter: test_gapi_sample*.py
Discovering python tests from: /home/atalaman/workspace/opencv-python/build
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/ml/misc/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/dnn/misc/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/features2d/misc/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/videoio/misc/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/calib3d/misc/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/objdetect/misc/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/stitching/misc/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/video/misc/python/test
    found 0 tests
Discovering python tests from: /home/atalaman/workspace/opencv-python/modules/gapi/misc/python/test
    found 17 tests
test_array_with_custom_type (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_add (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_addC (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_boundingRect (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_goodFeaturesToTrack (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_mean (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_size (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_sizeR (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_split3 (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_invalid_op (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_invalid_op_input (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_invalid_outMeta (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_opaq_with_custom_type (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_pipeline_with_custom_kernels (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_raise_in_kernel (test_gapi_sample_pipelines.gapi_sample_pipelines) ... /usr/lib/python3/dist-packages/apport/report.py:13: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import fnmatch, glob, traceback, errno, sys, atexit, locale, imp, stat
Traceback (most recent call last):
  File "/home/atalaman/workspace/opencv-python/modules/gapi/misc/python/test/test_gapi_sample_pipelines.py", line 540, in run
    raise Exception('Error')
Exception: Error
ok
test_raise_in_outMeta (test_gapi_sample_pipelines.gapi_sample_pipelines) ... Traceback (most recent call last):
  File "/home/atalaman/workspace/opencv-python/modules/gapi/misc/python/test/test_gapi_sample_pipelines.py", line 562, in outMeta
    raise 'Error'
TypeError: exceptions must derive from BaseException
ok
test_stateful_kernel (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok

----------------------------------------------------------------------
Ran 17 tests in 0.044s

OK

```

Reply:
Stack trace dump of crashes from gdb and Python 3.9:

<details>

```
...
Discovering python tests from: /home/alalek/projects/opencv/dev/modules/gapi/misc/python/test
    found 17 tests
test_array_with_custom_type (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_custom_op_add (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
ok
test_custom_op_addC (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
ok
test_custom_op_boundingRect (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
ok
test_custom_op_goodFeaturesToTrack (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
ok
test_custom_op_mean (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Expected cv::GProtoInputArgs for argument 'ins'
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Expected cv::GMat for argument 'out'
ok
test_custom_op_size (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
ok
test_custom_op_sizeR (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
ok
test_custom_op_split3 (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
ok
test_invalid_op (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_invalid_op_input (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_invalid_outMeta (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_opaq_with_custom_type (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok
test_pipeline_with_custom_kernels (test_gapi_sample_pipelines.gapi_sample_pipelines) ... [ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
[ WARN:0] global /home/alalek/projects/opencv/dev/modules/python/src2/cv2.cpp (110) emit_failmsg Bindings conversion failed: Argument 'normType' is required to be an integer
ok
test_raise_in_kernel (test_gapi_sample_pipelines.gapi_sample_pipelines) ... Traceback (most recent call last):
  File "/home/alalek/projects/opencv/dev/modules/gapi/misc/python/test/test_gapi_sample_pipelines.py", line 540, in run
    raise Exception('Error')
Exception: Error
ok
test_raise_in_outMeta (test_gapi_sample_pipelines.gapi_sample_pipelines) ... Traceback (most recent call last):
  File "/home/alalek/projects/opencv/dev/modules/gapi/misc/python/test/test_gapi_sample_pipelines.py", line 562, in outMeta
    raise NotImplementedError("outMeta ins't implemented")
NotImplementedError: outMeta ins't implemented
ok
test_stateful_kernel (test_gapi_sample_pipelines.gapi_sample_pipelines) ... ok

----------------------------------------------------------------------
Ran 17 tests in 0.023s

OK

Thread 1 "python3" received signal SIGSEGV, Segmentation fault.
0x00007fffe9898ba0 in PyFloatArrType_Type () from /home/alalek/penv3/lib64/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-x86_64-linux-gnu.so
(gdb) bt
#0  0x00007fffe9898ba0 in PyFloatArrType_Type () at /home/alalek/penv3/lib64/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-x86_64-linux-gnu.so
#1  0x00007ffff7d3ffdf in tupletraverse () at /lib64/libpython3.9.so.1.0
#2  0x00007ffff7d3fc65 in deduce_unreachable () at /lib64/libpython3.9.so.1.0
#3  0x00007ffff7d3f97d in collect () at /lib64/libpython3.9.so.1.0
#4  0x00007ffff7dc2a0e in collect_with_callback () at /lib64/libpython3.9.so.1.0
#5  0x00007ffff7df3a0e in PyGC_Collect () at /lib64/libpython3.9.so.1.0
#6  0x00007ffff7df3427 in Py_FinalizeEx () at /lib64/libpython3.9.so.1.0
#7  0x00007ffff7df379c in Py_Exit () at /lib64/libpython3.9.so.1.0
#8  0x00007ffff7df0f3b in handle_system_exit () at /lib64/libpython3.9.so.1.0
#9  0x00007ffff7df0c6d in PyErr_PrintEx () at /lib64/libpython3.9.so.1.0
#10 0x00007ffff7de962e in PyRun_SimpleFileExFlags () at /lib64/libpython3.9.so.1.0
#11 0x00007ffff7de6aa6 in Py_RunMain () at /lib64/libpython3.9.so.1.0
#12 0x00007ffff7db8f8d in Py_BytesMain () at /lib64/libpython3.9.so.1.0
#13 0x00007ffff7a9b1e2 in __libc_start_main () at /lib64/libc.so.6
#14 0x000055555555509e in _start ()
(gdb)
```

</details>

Reply:
@alalek Could you try it once again ?